### PR TITLE
flake/nixosSystem: pass the final lib to eval-config.nix

### PR DESCRIPTION
**Pull request by ncfavier**
Extracted from https://github.com/NixOS/nixpkgs/pull/157056. The `makeExtensible` fix isn't strictly necessary for this but seems reasonable regardless.

### CODE REVIEW

File: lib/fixed-points.nix [hunk 1]
```diff
@@ -107,7 +107,7 @@ rec {
   # Same as `makeExtensible` but the name of the extending attribute is
   # customized.
   makeExtensibleWithCustomName = extenderName: rattrs:
-    fix' rattrs // {
+    fix' (self: rattrs self // {
       ${extenderName} = f: makeExtensibleWithCustomName extenderName (extends f rattrs);
-   };
+    });
```
Comment (Infinisil) [inline lib/fixed-points.nix:112]: (see also https://github.com/NixOS/nixpkgs/pull/157056#discussion_r809325176 for why this is necessary)

**Comment by ncfavier** [comment]
Updated as per discussion in https://github.com/NixOS/nixpkgs/pull/157056#discussion_r816634635

**Comment by ncfavier** [comment]
Even if https://github.com/NixOS/nixpkgs/pull/157056 is too much, I think this should go through. Should we have a vote?

**Comment by roberth** [comment]
This looks innocent, but it is a significant shift in what `lib` means.

Currently:

`nixpkgsFlake.lib`: all the functions (and more) that the flake needs to expose, merged in with the Nixpkgs function library
`lib` module argument: the Nixpkgs function library

After this:

`nixpkgsFlake.lib`: the better of the two `lib`s that includes NixOS functions and possibly more
`lib` module argument: who knows

So this doesn't look like an unambiguous improvement.

**Pull request closed without merging**